src/fu-util.c

```diff
@@ -560,6 +617,227 @@ fu_util_install_prepared (FuUtilPrivate *priv, gchar **values, GError **error)
+		FuDevice *dev = g_ptr_array_index (results, i);
+		if (fu_device_has_flag (dev, FWUPD_DEVICE_FLAG_REPORTED))
+			continue;
+
+		json_builder_begin_object (builder);
+
+		/* identify different devices */
+		json_builder_set_member_name (builder, "DeviceId");
+		json_builder_add_string_value (builder, fu_device_get_id (dev));
+
+		/* identify the firmware used */
+		json_builder_set_member_name (builder, "FirmwareId");
+		json_builder_add_string_value (builder, fu_device_get_firmware_id (dev));
+
+		/* set the error state of the report */
+		json_builder_set_member_name (builder, "UpdateState");
+		json_builder_add_int_value (builder, fu_device_get_update_state (dev));
+		if (fu_device_get_update_error (dev) != NULL) {
+			json_builder_set_member_name (builder, "UpdateError");
+			json_builder_add_string_value (builder, fu_device_get_update_error (dev));
+		}
+
+		/* map back to the dev type on the LVFS */
+		json_builder_set_member_name (builder, "Guid");
+		json_builder_add_string_value (builder, fu_device_get_guid_default (dev));
+
+		/* to know what plugin tried to handle the dev */
+		json_builder_set_member_name (builder, "FwupdVersion");
+		json_builder_add_string_value (builder, fu_device_get_metadata (dev, "FwupdVersion"));
+		json_builder_set_member_name (builder, "Plugin");
+		json_builder_add_string_value (builder, fu_device_get_plugin (dev));
+
+		/* report what we're trying to update *from* and *to* */
+		json_builder_set_member_name (builder, "Version");
+		json_builder_add_string_value (builder, fu_device_get_version (dev));
+		json_builder_set_member_name (builder, "VersionNew");
+		json_builder_add_string_value (builder, fu_device_get_version_new (dev));
+
+		/* to know the state of the dev we're trying to update */
+		json_builder_set_member_name (builder, "Flags");
+		json_builder_add_int_value (builder, fu_device_get_flags (dev));
+
+		/* to know when the update tried to happen, and how soon after boot */
+		json_builder_set_member_name (builder, "Created");
+		json_builder_add_int_value (builder, fu_device_get_created (dev));
+		json_builder_set_member_name (builder, "Modified");
+		json_builder_add_int_value (builder, fu_device_get_modified (dev));
+
+		json_builder_end_object (builder);
+	}
+	json_builder_end_array (builder);
+	json_builder_end_object (builder);
+
+	/* export as a string */
+	json_root = json_builder_get_root (builder);
+	json_generator = json_generator_new ();
+	json_generator_set_pretty (json_generator, TRUE);
+	json_generator_set_root (json_generator, json_root);
+	data = json_generator_to_data (json_generator, NULL);
+	if (data == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "Failed to convert to JSON string");
+		return FALSE;
+	}
+
+	/* ask for permission */
+	fu_util_print_data (_("Target"), report_uri);
+	fu_util_print_data (_("Payload"), data);
+	g_print ("%s [Y|n]: ", _("Proceed with upload?"));
```

Comment:
I'm still not crazy about this being opt-in, but if it has to be opt in and nag, rather than just Y | N for this, how about two questions.
1. Y | N for the upload
2. If Y - "automatically upload results in the future?"

If they say yes to automatically report, then use a new method on the daemon to update `/etc/fwupd/client.conf`.

Reply:
I did some unofficial polling here: https://plus.google.com/u/0/+RichardHughes/posts/e97bZz8Lbum -- 114 responses, 77% of people happy with automated collection :/ I'd be happy to add the automatic upload question too, although that needs a bit more thinking about. For GNOME Software, we'd have a privacy slider, default on, and that's going to cover the vast majority of users.

Reply:
Yeah I do forget that most people who do their updates probably do it from gnome-software rather than fwupdmgr.  A slider there is probably sufficient to cover the majority of users.